game-core/src/test/java/games/strategy/triplea/delegate/DiceRollTest.java

```diff
@@ -63,19 +63,23 @@ void testSimple() {
         .thenAnswer(withValues(1)); // infantry attack does not hit at 1 (0 based)
     // infantry defends
     final DiceRoll roll =
-        DiceRoll.rollDice(infantry, true, russians, bridge, battle, "", territoryEffects, null);
+        DiceRoll.rollDice(
+            infantry, true, russians, bridge, battle, "", territoryEffects, null, infantry);
     assertThat(roll.getHits(), is(1));
     // infantry
     final DiceRoll roll2 =
-        DiceRoll.rollDice(infantry, true, russians, bridge, battle, "", territoryEffects, null);
+        DiceRoll.rollDice(
+            infantry, true, russians, bridge, battle, "", territoryEffects, null, infantry);
     assertThat(roll2.getHits(), is(0));
     // infantry attacks
     final DiceRoll roll3 =
-        DiceRoll.rollDice(infantry, false, russians, bridge, battle, "", territoryEffects, null);
+        DiceRoll.rollDice(
+            infantry, false, russians, bridge, battle, "", territoryEffects, null, infantry);
     assertThat(roll3.getHits(), is(1));
     // infantry attack
     final DiceRoll roll4 =
-        DiceRoll.rollDice(infantry, false, russians, bridge, battle, "", territoryEffects, null);
+        DiceRoll.rollDice(
```

Comment:
Would a method overload be cleaner to avoid passing null parameter when the last parameter is unused?

Reply:
Possibly. I may add some overloads to this but wanted to ensure I updated all the necessary places that needed updated first and not miss any.

Reply:
Updated.